solarkraft/src/io/itf.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * [Apache-2.0](https://github.com/freespek/solarkraft/blob/main/LICENSE)
+ */
+import { byte, getFullType, Value } from '../state/value.js'
+import { State } from '../state/state.js'
+
+const META_FIELD: string = '#meta'
+const VAR_TYPES_FIELD: string = 'varTypes'
+const VARS_FIELD: string = 'vars'
+const DESCRIPTION_FIELD: string = 'description'
+const FORMAT_FIELD: string = 'format'
+const FORMAT_DESCRIPTION_FIELD: string = 'format-description'
+const BIG_INT_FIELD: string = '#bigint'
+const MAP_FIELD: string = '#map'
+const UNSERIALIZABLE_FIELD: string = '#unserializable'
+const STATES_FIELD: string = 'states'
+
+const UNSERIALIZABLE = { [UNSERIALIZABLE_FIELD]: true }
+
+// Wraps bigint or byte literals in the ITF #bigint wrapper
+function numWrap(b: bigint | byte) {
+    return { [BIG_INT_FIELD]: b.toString() }
+}
+
+// Transforms the internal representation into the Apalache ITF format for serialization
+export function valToITF(v: Value) {
+    switch (v.type) {
+        case 'bool':
+            return v.val
+        case 'u32':
+        case 'i32':
+        case 'u64':
+        case 'i64':
+        case 'u128':
+        case 'i128':
+            return numWrap(v.val)
+        case 'symb':
+        case 'addr':
+            return v.val
+        case 'arr':
+            // array of 0/1 s, must be wrapped
+            return v.val.map(numWrap)
```

Comment:
Yes, I did it for consistency, but for `Bytes` specifically, we could deviate from ITF and just use the JS numbers, since it's only 0s and 1s, b/c I do agree that it's a bit clunky.

Reply:
If it's an array of zeroes and ones, why would not we just wrap it as a string, e.g., by introducing a new kinds of a field `{ "#bytes": "101" }`, or by just using an array of numbers, as you are suggesting. Actually, ITF does not prohibit JS numbers from being used for the small ranges.

Reply:
IIRC we had a discussion about standardizing this in Apalache where _every_ value was `#bigint`. But yeah, we don't have to 100% copy ITF here

Reply:
Why do we encode a `bytes[]` as bits in the first place?

Reply:
because I'm illiterate apparently

Reply:
Is this question resolved? I am not sure I understand the last comment.

Reply:
Yes, I fixed it. `bytes = number` now, and the validations check for `2^8`